File: src/main/java/org/team1/nbe1_3_team01/domain/attendance/service/port/AttendanceRepository.java
package org.team1.nbe1_3_team01.domain.attendance.service.port;

import java.util.List;
import java.util.Optional;
import org.team1.nbe1_3_team01.domain.attendance.entity.Attendance;

public interface AttendanceRepository {

    Optional<Attendance> findById(Long id);

    List<Attendance> findByUserId(Long userId);

    List<Attendance> findAll();

    Attendance save(Attendance attendance);

    void deleteById(Long id);
}

File: src/main/java/org/team1/nbe1_3_team01/domain/attendance/service/port/AttendanceRepository.kt
package org.team1.nbe1_3_team01.domain.attendance.service.port

import org.team1.nbe1_3_team01.domain.attendance.entity.Attendance

interface AttendanceRepository {

    fun findById(id: Long): Attendance?

    fun findByUserId(userId: Long): List<Attendance>

    fun findAll(): List<Attendance>

    fun save(attendance: Attendance): Attendance

    fun deleteById(id: Long)
}